components/script/dom/htmlvideoelement.rs

```diff
@@ -280,11 +316,32 @@ impl VirtualMethods for HTMLVideoElement {
     fn attribute_mutated(&self, attr: &Attr, mutation: AttributeMutation) {
         self.super_type().unwrap().attribute_mutated(attr, mutation);
 
-        if let Some(new_value) = mutation.new_value(attr) {
-            if attr.local_name() == &local_name!("poster") {
-                self.fetch_poster_frame(&new_value, CanGc::note());
-            }
-        }
+        match *attr.local_name() {
+            local_name!("poster") => match mutation {
+                AttributeMutation::Set(_) => {
+                    if let Some(new_value) = mutation.new_value(attr) {
+                        self.fetch_poster_frame(&new_value, CanGc::note())
+                    }
+                },
+                AttributeMutation::Removed => {
+                    self.set_intrinsic_size();
+                    self.htmlmediaelement.clear_current_frame();
+                },
+            },
+            local_name!("src") => {
+                if matches!(mutation, AttributeMutation::Removed) {
```

Comment:
```suggestion
                if mutation.is_removal() {
```

But shouldn't this also happen when `src` is set to a new value?